jobrunner/executors/local.py

```diff
@@ -395,9 +395,26 @@ def finalize_job(job_definition):
 
     # First get the user-friendly message for known database exit codes, for jobs
     # that have db access
-    message = None
+    unmatched_hint = None
 
-    if exit_code == 137 and job_definition.cancelled:
+    if exit_code == 0 and unmatched_patterns == [] and outputs is not None:
+        message = "Completed successfully"
+
+    elif (
+        exit_code == 0
+        and unmatched_patterns is not None
+        and unmatched_outputs is not None
+    ):
+        message = "\n  No outputs found matching patterns:\n - {}".format(
+            "\n   - ".join(unmatched_patterns)
+        )
+        unmatched_hint = (
```

Comment:
I was wondering if it was worth having a separate `unmatched_hint` variable for this, or whether it might be simpler to just append this to the message if we have unmatched_outputs.

However, doing it this way means we don't send file names up to job-server, which is the currently policy, even if its probably going to change some point soon.

So on balance, I do think that a separate hint variable is worth it, nice work.